cpp/arcticdb/column_store/chunked_buffer.hpp

```diff
@@ -26,6 +26,14 @@
 
 namespace arcticdb {
 
+/*
+ * ChunkedBufferImpl is an untyped buffer that is composed of blocks of data that can be either regularly or
+ * irregularly sized, with optimizations for both cases as well as the case where there is a single contiguous
```

Comment:
There are optimizations for:

- A single block
- Multiple blocks that are all regularly sized
- Multiple blocks that are regularly sized up until a point, and irregularly sized after that point.

If the blocks are all irregularly sized there's nothing we can do to optimize.